presto-tests/src/main/java/com/facebook/presto/tests/QueryAssertions.java

```diff
@@ -51,12 +56,26 @@ private QueryAssertions()
     {
     }
 
-    public static void assertUpdate(QueryRunner queryRunner, Session session, @Language("SQL") String sql, OptionalLong count)
+    public static void assertUpdate(QueryRunner queryRunner, Session session, @Language("SQL") String sql, OptionalLong count, Optional<Consumer<Plan>> planAssertion)
     {
         long start = System.nanoTime();
-        MaterializedResult results = queryRunner.execute(session, sql);
+        MaterializedResult results;
+        Plan queryPlan;
+        if (planAssertion.isPresent()) {
+            MaterializedResultWithPlan resultWithPlan = queryRunner.executeWithPlan(session, sql, WarningCollector.NOOP);
+            queryPlan = resultWithPlan.getQueryPlan();
+            results = resultWithPlan.getMaterializedResult().toTestTypes();
+        }
+        else {
+            queryPlan = null;
+            results = queryRunner.execute(session, sql);
+        }
         log.info("FINISHED in presto: %s", nanosSince(start));
 
+        if (planAssertion.isPresent()) {
```

Comment:
We can also write as
```java
        planAssertion.ifPresent(assertion -> assertion.accept(queryPlan));
```

ditto for line 150.

I don't have strong option here, though. Given we anyway have to have the if-statement from line 64-72

Reply:
I'll leave this as is.